### ISSUE TITLE
bug: Slowdown / non-standard browser load

### ISSUE THREAD

**Issue by Stepan-H** [twofas/2fas-browser-extension#67]
### Browser info

Mozilla Firefox 123.0 64bit

### Browser extension version

1.6.3

### Bug description

Extreme and long CPU using during/after page loading. Firefox offers to disable the add-on because of this.
So far, this is only happening to me on the company's intraweb pages.

In Firefox I got this from the about:processes, see image attachment. Red line shows "15,8s Jank - event procesing delay"
![2FAS issue 202303011511](https://github.com/twofas/2fas-browser-extension/assets/31160576/461f0892-cb7e-4a4a-aa1c-dbd5bdac485e)

### Acknowledgements

- [X] This issue is not a duplicate of an existing bug report.
- [X] I understand that security vulnerabilities should be reported to security@2fas.com instead of on GitHub.
- [X] I have chosen an appropriate title.
- [x] All requested information has been provided properly.

**Comment by GrzegorzZajac000** [comment]
I will be fixed with v1.6.4

**Comment by brandoningli** [comment]
Just want to document that I'm experiencing this same issue on the AWS console. Will keep an eye out for the v1.6.4 rollout!

**Comment by last-partizan** [comment]
This means, 2fa extension runs on every page without user interaction?

Maybe better way to solve this - is to run extension only after user clicks extension icon?

**Comment by SAJewers** [comment]
Getting this searching Steam as well.

**Comment by zirdum** [comment]
I've noticed the same issue with slowdown switching between sections within the "You Need a Budget" site, but using the Chrome browser (v122.0.6261.112) on a macbook. When I disable the extension it runs normally.

**Comment by iGlitch** [comment]
Similar experience as aforementioned.

Any way to make an exclude domain list?
Similar to what exists for the auto-submit exclude list... but it would be just complete exclude from running at all domain list.

**Comment by KobeW50** [comment]
> Similar experience as aforementioned (on TheLounge IRC)
> 
> Any way to make an exclude domain list?
> Similar to what exists for the auto-submit exclude list... but it would be just complete exclude from running at all domain list.

The browser extension itself doesn't support this, but some browsers may have such a feature to manage which browser extensions can run on which pages.

**Comment by Enivex** [comment]
Also happening on https://typst.app/docs/reference/symbols/sym/

**Comment by GrzegorzZajac000** [comment]
v1.6.4 was uploaded to stores. Should be available in next hours.

**Comment by Enivex** [comment]
> v1.6.4 was uploaded to stores. Should be available in next hours.

Is it expected to fix these issues?

**Comment by FWest98** [comment]
While the performance in 1.6.4 is better, it is still very choppy compared to running without the plugin (and compared to previous versions).

**Comment by GrzegorzZajac000** [comment]
@FWest98 We know the problem wasn't solved perfectly, but we wanted to solve the problem as quickly as possible. I'm already working on version 1.7.0, which will slightly change the way it works in the direction that @last-partizan described.

**Comment by Stepan-H** [comment]
Thank you. It's better than before, but as you wrote, It's not best solution. Browser does not stuck and does not show turn-off addon offer, but there are stil remain CPU loading problem. I look forward to a revised version 1.7. and wish you all the best with your rework.

**Comment by Enivex** [comment]
I can confirm that https://typst.app/docs/reference/symbols/sym/ works much better now

**Comment by gcstang** [comment]
Same issue on 1.6.4 for FF 124.0.1 (64-bit) on macOS 13.6.4 (22G513)
Firefox notified me 2FAS extension is slowing down all page loads and had to disable it (3/25/2024)
I move from Chrome thinking it was the browser but it seems like it's the same exact issue in FF, disabling fixes the browser load times.

**Comment by Andrew15-5** [comment]
YouTube Music is also affected. Previously only with this extension enabled it would spontaneously start loading CPU after some time until I reload the page. Today, the Firefox even gave me the notification and the player didn't play the next song until I pressed "Stop" button.

![image](https://github.com/twofas/2fas-browser-extension/assets/37143421/5c6e6bbb-3cf2-4c3b-b83b-e2f14d8cd3c1)

**Comment by GrzegorzZajac000** [comment]
@mryx007 Previous versions contained other bugs, and slowdown has always occurred in some form. Version v1.7.0, which introduces a new mechanism and eliminates this problem, is already in our internal tests. It will be publicly available soon.

**Comment by GrzegorzZajac000** [comment]
v1.7.0 was uploaded to stores. Should be available in next hours.
Thanks one more time for @last-partizan for comment. We implemented this idea by adding a few more changes ;)
